src/Symfony/Component/Workflow/Transition.php

```diff
@@ -39,18 +46,56 @@ public function getName(): string
     }
 
     /**
+     * @deprecated since Symfony 7.4, use getFromArcs() instead
+     *
      * @return string[]
      */
     public function getFroms(): array
```

Comment:
I'm not a big fan of deprecating `getFroms()`. This feels like a new and more advanced concern leaking into simple use cases.
Instead, what about doing this?

```suggestion
    public function getFroms(/* bool $weighted = false*/): array
```

When `$weighted` is true, we return the places as keys and their weight as values.
Would that work? If yes, we could even remove the `Arc` class and do this with arrays only, for simplicity.

Reply:
Arc class is important to me. Class are much better than array:
* consume less memory
* better property (key) discoverabiliy
* doesn't need extra PHPDoc
* better handling by static analysis.